### PR TITLE
chore(experiments): set up e2e test

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Changes
Set up e2e test for experiments. Mocking `/experiment` responses actually does solve our problem around licenses 🙌 

The mocked response in `new-experiment.json` diverges from the random data generated during the test. This is not an issue for this test, but we might want to bring them in sync if we do a more complicated flow later on (e.g. create experiment - check created experiment in the list).

This test needs to be extended, but want to get some preliminary feedback first.

## How did you test this code?
🧪

### CODE REVIEW

File: cypress/support/e2e.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ beforeEach(() => {
                 'surveys-results-visualizations': true,
                 'auto-redirect': true,
                 notebooks: true,
+                experimentation: true,
```
Comment (neilkakkar) [inline cypress/support/e2e.ts:37]: don't think any experiments flag are still in use, are they?

**Review by neilkakkar — APPROVED** [review]
Very nice, this is good for now! :)

I'd want to not depend on mocking everything later on, and let the API do the right things, but let's not block on it and write these tests first

**Comment by neilkakkar** [comment]
Noticing the permissions depend on the deprecated available_features -> https://github.com/PostHog/posthog/blob/master/posthog/permissions.py#L229 -> @raquelmsmith should this be moved to `available_product_features` ? Or is it fine as is, it's backwards compatible?

**Comment by raquelmsmith** [comment]
Please move to `available_product_features`!

File: cypress/fixtures/api/experiments/user.json
Comment (raquelmsmith) [inline cypress/fixtures/api/experiments/user.json:1]: does one of the existing mocked users not work for this?

File: cypress/support/e2e.ts (hunk 1 above)
Comment (jurajmajerik) [reply cypress/support/e2e.ts:37]: true

File: cypress/fixtures/api/experiments/user.json
Comment (jurajmajerik) [reply cypress/fixtures/api/experiments/user.json:1]: I'm not sure what the intent is with the other user mocks, so just wanted to keep this isolated.

Like here, this is probably some legacy stuff, I'm not even sure when it gets executed:
```
    if (Cypress.spec.name.includes('Premium')) {
        cy.intercept('/api/users/@me/', { fixture: 'api/user-enterprise' })
```

Anyway, mocking the user for experiments is just temporary, as we plan on removing Experiments from Enterprise features. Need to think this through more, but could be something like "first 3 experiments free, then need to pay".

**Comment by jurajmajerik** [comment]
@raquelmsmith changing [this line](https://github.com/PostHog/posthog/blob/master/posthog/permissions.py#L229) to `available_product_features` breaks existing tests 😬 changed it back to `available_features`

**Pull request merged**